code-challenges.test.js

```diff
@@ -14,43 +14,120 @@
+        const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    const letterA = "a"
+    // Expected output: ["Apple", "Banana", "Orange"]
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    const letterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+        expect(arrayContains(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+        expect(arrayContains(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+// make a function that have two parameters
+// use a filter method to filter out words that includes the character given
+// also use the lowercase method so that every letter will be equal to the given arguments
+// return the filtered words
+function arrayContains(arr, char) {
+    return arr.filter(a => {
+        return a.toLowerCase().includes(char)
+    })
+}
+
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
-
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-
+describe("fullHouse", function(){
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house", function(){
+        const hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        const hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        const hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
+        expect(fullHouse(hand1)).toEqual(true)
+        expect(fullHouse(hand2)).toEqual(false)
+        expect(fullHouse(hand3)).toEqual(false)
+        
+    })
+})
 
 // b) Create the function that makes the test pass.
+
+//create a function name full house that takes in an array
+// make a variable name noDuplicate that revmove duplicate from the array
+// use the map method with a conditional statement where you add 1 to first variable if array is equals to first index and add 1 to second variable if array is equals to seconde index
+// make a new conditional statement that if the first var is equals three and second var is equals two or if first var is equals two and second var is equals three, then it is a fullhouse
+// return true if full house and false if not
+function fullHouse(arr) {
+    let noDup = [...new Set(arr)]
+    let first = 0
+    let second = 0
+    arr.map(a => {
+        if (a === noDup[0]) {
+            first++
+        } if (a === noDup[1]) {
+            second++
+        }
+    })
+    if (first === 3 && second === 2){
+        return true
+    } else if (first === 2 && second === 3) {
+        return true
+    } else {
+        return false
+    }
+}
```

Comment:
Nice use of new Set with the spread operator!  Love that you used higher-order functions in this.  Great work!